learn/advanced/filtering_and_faceted_search.md

```diff
@@ -45,29 +33,37 @@ Suppose you have a collection of movies containing the following fields:
 ]
 ```
 
-If you want to filter results based on the `director` and `genres` attributes, you must add them to the [`filterableAttributes` list](/reference/api/settings.md#filterable-attributes):
+If you want to filter results based on the `director` and `genres` attributes, you must add them to the `filterableAttributes` list:
 
 <CodeSamples id="faceted_search_update_settings_1" />
 
-## Using filters
+**This step is mandatory and cannot be done at search time**  because Meilisearch needs to prepare an index before you can use filters. Updating `filterableAttributes` may take a significant amount of time depending on your dataset size.
+
+::: note
+By default, `filterableAttributes` is empty. Filters do not work without first explicitly adding attributes to the `filterableAttributes` list.
+:::
+
+Filters work with numeric and string values. Empty fields or fields containing an empty array will be ignored.
 
-Once you have configured `filterableAttributes`, you can start using [the `filter` search parameter](/reference/api/search.md#filter). Search parameters are added to at search time, that is, when a user searches your dataset.
+Filters do not work with [`NaN`](https://en.wikipedia.org/wiki/NaN) and infinite values such as `inf` and `-inf` as they are [not supported by JSON](https://en.wikipedia.org/wiki/JSON#Data_types). It is possible to filter infinite and `NaN` values if you parse them as strings, except when handling [`_geo` fields](/learn/advanced/geosearch.md#preparing-documents-for-location-based-search).
+
+## Filter basics
+
+Once you have configured `filterableAttributes`, you can use [the `filter` search parameter](/reference/api/search.md#filter). The `filter` search parameter refines search results by selecting documents matching the given filter and running the search query only on those documents.
+
+::: warning
+[Synonyms](/learn/configuration/synonyms.md) don't apply to filters. Meaning, if you have `SF` and `San Francisco` set as synonyms, filtering by `SF` and `San Francisco` will show you different results.
+:::
```

Comment:
This does not feel like the best place to put this information. It is not basic and breaks up the flow. I think it might be better in the section that's currently titled "Using filters", which seems simply by its placement further down in the guide to be more advanced, and thus a better place to list exceptions etc..